Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -243,22 +254,24 @@ - (void)deleteTypedCharacter:(unichar)deletedChar withCharacterNowPrecedingCurso
 
 - (void)cursorMovedWithCharacterNowPrecedingCursor:(unichar)c {
     NSCharacterSet *whitespaceSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
+    NSCharacterSet *punctuationCharacterSet = [NSCharacterSet punctuationCharacterSet];
```

Comment:
Can we combine these using `NSMutableCharacterSet formUnionWithCharacterSet:`?  This will eliminate half of these checks. (here and elsewhere in this pr)

Reply:
> Can we combine these using `NSMutableCharacterSet formUnionWithCharacterSet:`? This will eliminate half of these checks. (here and elsewhere in this pr)

We can create a character set with union of punctuation character set and whitespace character set. But, it would require us to create a mutable character set(and union it with both sets) and its copy. This copy would only be used once to set enum value to character type as this type is used further unless we need a specific check of whitespace or punctuation. Additionally, we would have to create whitespace set or punctuation character set or both for few specific checks. This would create multiple character sets with minimal use in multiple methods. 

Please let us know if it would still be better to have a union of both sets.

Reply:
Can you save to a readonly class property that's dispatch_once'd then?